### ISSUE TITLE
Installation Errors - HACS Install

### ISSUE THREAD

**Issue by townsmcp** [fano0001/home-assistant-mazda#2]
Core - 2024.6.4
Supervisor - 2024.06.2
Operating System - 12.4
Frontend - 20240610.1

Issues when trying to install the latest package:

1. Cleared out HA of previous Mazda installs, confirmed that `/homeassistant/custom_components/mazda` did not exist.
2. Rebooted HA
3. Added this repo to HACS
4. Rebooted
5. Attempt to add the integration - unable to locate integration
<img width="594" alt="image" src="https://github.com/user-attachments/assets/a75ee03e-79ed-42f2-bb75-5c2106e2d8f4">

I have however observed that the integration is residing in `/homeassistant/custom_components/mazda/mazda`
<img width="396" alt="image" src="https://github.com/user-attachments/assets/bca5c14c-a524-49b0-8c64-d3bc021b6ddb">

Moving the content of `/homeassistant/custom_components/mazda/mazda` to `/homeassistant/custom_components/mazda`, restarting HA and trying to add the integration again does show the integration in the list (in pic above) however there is now a config flow error `Config flow could not be loaded: {"message":"Invalid handler specified"}'`
<img width="629" alt="image" src="https://github.com/user-attachments/assets/a4e2f06e-5254-4f09-a7bb-ffe11d1beddb">

I then tried deleting everything, rebooting HA and downloading the files manually and adding to `/homeassistant/custom_components/mazda` however I still get the same error message about the config flow. Any suggestions on what I am doing wrong to get this working?

**Comment by townsmcp** [comment]
Just to update this, so that I have something working for the time being I installed runDMCA version but modified the files mentioned from 15th Feb documented here https://github.com/fano0001/home-assistant-mazda/commit/de25ecd98e6eb6d27b15530f75ce2c3cabf8dce1 and I have the `Mazda CX-60 Remaining changing time (AC)` sensor working 😄 

When I added all code changes from v1.1.0 on May 7th from https://github.com/fano0001/home-assistant-mazda/commit/d62889b149a1bd4f972ce19867f5cb3aa8c11db8  it all breaks and errors on reboot:

```
Logger: homeassistant.setup
Source: setup.py:320
First occurred: 9:38:03 pm (1 occurrences)
Last logged: 9:38:03 pm

Setup failed for custom integration 'mazda': Unable to import component: Exception importing custom_components.mazda
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1052, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 131, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/mazda/__init__.py", line 32, in <module>
    from .pymazda.client import Client as MazdaAPI
  File "/config/custom_components/mazda/pymazda/client.py", line 209
    "drivingRangeBevKm" charge_info.get("BatRemDrvDistKm"),
                      ^
SyntaxError: ':' expected after dictionary key

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 992, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1063, in _get_component
    raise ImportError(f"Exception importing {self.pkg_path}") from err
ImportError: Exception importing custom_components.mazda

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1052, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 131, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/mazda/__init__.py", line 32, in <module>
    from .pymazda.client import Client as MazdaAPI
  File "/config/custom_components/mazda/pymazda/client.py", line 209
    "drivingRangeBevKm" charge_info.get("BatRemDrvDistKm"),
                      ^
SyntaxError: ':' expected after dictionary key

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 320, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1012, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1004, in async_get_component
    comp = self._get_component()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1063, in _get_component
    raise ImportError(f"Exception importing {self.pkg_path}") from err
ImportError: Exception importing custom_components.mazda
```

**Comment by townsmcp** [comment]
Found the issue!
Line 209 of custom_components/mazda/pymazda/client.py has a typo, it is missing a : between the definition of name and the sensor get
Original:
```
	                "drivingRangeBevKm" charge_info.get("BatRemDrvDistKm"),
```
Change to:
```
	                "drivingRangeBevKm": charge_info.get("BatRemDrvDistKm"),
```

**Comment by townsmcp** [comment]
I think I have identified why installs are going into sub directory, see https://github.com/hacs/integration/issues/931 - I think it is because th `mazda` domain has already been used. Maybe update `manifest.json` `"domain": "mazda"` to `"domain": "mazda_vehicle"` ? I did try removing integration, HACS integration repo link and fully restarting HA instance rather than just a reboot but that didnt work either

**Comment by fano0001** [comment]
i will try to fix the installation

**Comment by fano0001** [comment]
The issue with the installation is fixed in version 1.3.0
